app/models/involved_civilian.rb

```diff
@@ -48,6 +50,10 @@ class InvolvedCivilian < InvolvedPerson
     validates :firearm_type, presence: true, subset: { in: FIREARM_TYPES }, if: :firearm?
   end
 
+  def k12_setting?
+    incident.try(:general_info).try(:on_k12_campus)
```

Comment:
Shouldn't need `.try` here as general info should ALWAYS exist if you're creating a civilian, and `on_k12_campus` is a mandatory field